kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks.sh

```diff
@@ -265,142 +272,164 @@ function createYamlFiles {
-    echo Connencting to Azure Kubernetes Service.
-    az aks get-credentials --resource-group $azureResourceGroupName --name $aksClusterName
+  echo Creating Azure Kubernetes Service ${aksClusterName}
+  az aks create --resource-group $azureResourceGroupName \
+  --name $aksClusterName \
+  --vm-set-type VirtualMachineScaleSets \
+  --node-count ${azureKubernetesNodeCount} \
+  --generate-ssh-keys \
+  --nodepool-name ${azureKubernetesNodepoolName} \
+  --node-vm-size ${azureKubernetesNodeVMSize} \
+  --location $azureLocation \
+  --service-principal $azureServicePrincipalAppId \
+  --client-secret $azureServicePrincipalClientSecret
+
+  # Connect to AKS cluster
+  echo Connencting to Azure Kubernetes Service.
+  az aks get-credentials --resource-group $azureResourceGroupName --name $aksClusterName
 }
 
 function createFileShare {
-    # Create a storage account
-    echo Check if the storage account ${storageAccountName} exists.
-    ret=$(az storage account check-name --name ${storageAccountName})
-    nameAvailable=$(echo "$ret" | grep "nameAvailable" | grep "false")
-    if [ -n "$nameAvailable" ];then
-      echo $ret
-      fail "Storage account ${aksClusterName} is unavaliable."
-    fi
+  # Create a storage account
+  echo Check if the storage account ${storageAccountName} exists.
+  ret=$(az storage account check-name --name ${storageAccountName})
+  nameAvailable=$(echo "$ret" | grep "nameAvailable" | grep "false")
+  if [ -n "$nameAvailable" ]; then
+    echo $ret
+    fail "Storage account ${aksClusterName} is unavaliable."
+  fi
 
-    echo Creating Azure Storage Account ${storageAccountName}.
-    az storage account create \
-    -n $storageAccountName \
-    -g $azureResourceGroupName \
-    -l $azureLocation \
-    --sku ${azureStorageAccountSku}
-
-    # Export the connection string as an environment variable, this is used when creating the Azure file share
-    export azureStorageConnectionString=$(az storage account show-connection-string \
-    -n $storageAccountName -g $azureResourceGroupName -o tsv)
-
-    # Create the file share
-    echo Check if file share exists
-    ret=$( az storage share exists --name ${azureStorageShareName} --account-name ${storageAccountName} --connection-string $azureStorageConnectionString | grep "exists" | grep false)
-    if [[ "$ret" == "true" ]];then
-      fail "File share name  ${azureStorageShareName} is unavaliable."
-    fi
+  echo Creating Azure Storage Account ${storageAccountName}.
+  az storage account create \
+  -n $storageAccountName \
+  -g $azureResourceGroupName \
+  -l $azureLocation \
+  --sku Premium_LRS \
+  --kind FileStorage \
+  --https-only false \
+  --default-action Deny
+
+  echo Creating Azure NFS file share.
+  az storage share-rm create \
+  --resource-group $azureResourceGroupName \
+  --storage-account $storageAccountName \
+  --name ${azureStorageShareName} \
+  --enabled-protocol NFS \
+  --root-squash NoRootSquash \
+  --quota 100
```

Comment:
Minor: If you want to aid with tracing/debugging throughout, maybe consider pattern like:
```
local cmd="az storage share-rm create ... --quota 100"
echo "Running cmd '$cmd'"
$cmd
```
Since az is used so frequently, you could make a function that does this:
```
runCmd () {
  echo "about to run '$*'"
  $*
}

runCmd ls /tmp/*.log ~/*.sh
```

Reply:
This is a very useful idea to deal with the commands in a same pattern.

`local cmd="..."` get the output message, while we want to keep the progress message during the az command is running, I  will keep the script as it's.